### PR TITLE
Employee profile page

### PR DESCRIPTION
**Pull request by INeizvestnaya**
https://innowise-group.atlassian.net/wiki/spaces/~9373156/pages/2617311360/Employee+Profile+Page

### CODE REVIEW

File: src/components/EmployeeTabs/EmployeeTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        TabIndicatorProps={{ style: { background: '#c63031' } }}
+      >
+        {tabs.map(({ label, path }) => (
+          <Tab
+            key={label}
+            label={label}
+            component={NavLink}
+            value={path}
+            to={path}
+            sx={{ fontWeight: 'bold', paddingX: 6 }}
```
Comment (dzianislysionak) [inline src/components/EmployeeTabs/EmployeeTabs.tsx:50]: Would be nice to use same approach for styling. styled or sx in whole project
image

File: src/components/UserMenu/UserMenu.tsx [hunk 2]
```diff
@@ -33,14 +34,26 @@ export const UserMenu = () => {
     authService.clearStorage();
   };
 
+  if (loading) {
```
Comment (dzianislysionak) [inline src/components/UserMenu/UserMenu.tsx]: For me it candidate to create reusable component
`<Preloader isLoading={isLoading} error={error}>
.....
<Preloader />`
If it is not clear let me know in chat

File: src/pages/EmployeeCvsPage/EmployeeCvsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import EmployeeTabs from '../../components/EmployeeTabs';
```
Comment (dzianislysionak) [inline src/pages/EmployeeCvsPage/EmployeeCvsPage.tsx:1]: What the idea of this component? For me it is looks like you just reexport EmployeeTabs component

File: src/pages/EmployeeLanguagesPage/EmployeeLanguagesPage.tsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import EmployeeTabs from '../../components/EmployeeTabs';
```
Comment (dzianislysionak) [inline src/pages/EmployeeLanguagesPage/EmployeeLanguagesPage.tsx:1]: See comment above

File: src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
+import { useMutation, useQuery, useReactiveVar } from '@apollo/client';
+import { Box } from '@mui/material';
+import { ChangeEvent } from 'react';
+import { useLocation } from 'react-router-dom';
+
+import EmployeeTabs from '../../components/EmployeeTabs';
```
Comment (dzianislysionak) [inline src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx:5]: Just a suggestion Would be nice to set aliases for avoid import ... from '../../components/'
With aliases will look import ,,, from '@componts'

File: src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+            base64,
+            size: file.size,
+            type: file.type
+          }
+        }
+      });
+    }
+  };
+
+  if (
```
Comment (dzianislysionak) [inline src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx]: Just reminder. You can use <Preloader>

**Review by dzianislysionak — COMMENTED** [review]
Good job!!!!

File: src/components/UserProfileHeader/UserProfileHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  deleteAvatar: () => void;
+  uploadAvatar: (event: ChangeEvent<HTMLInputElement>) => void;
+}
+
+const getUserInitials = (user?: IUser) => {
+  if (!user) {
+    return '';
+  }
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
```
Comment (mikepetrusha) [inline src/components/UserProfileHeader/UserProfileHeader.tsx:44]: It is better to disable esling rules in the eslintrc file if you dont need them.

File: src/components/UserProfileHeader/UserProfileHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+        </Box>
+        {ableToEdit && (
+          <Box onClick={fileInputClick}>
+            <StyledTypography variant="button" fontSize={20}>
+              <StyledFileUploadIcon />
+              Upload avatar image
+            </StyledTypography>
+            <Typography color={theme.palette.lightGrey.dark}>
+              png, jpg or gif no more than 0.5MB
+            </Typography>
```
Comment (mikepetrusha) [inline src/components/UserProfileHeader/UserProfileHeader.tsx]: User should be able to drag and drop files.

**Review by mikepetrusha — COMMENTED** [review]
image
Please add the ability to navigate to your own profile page by clicking this button

File: src/pages/EmployeeCvsPage/EmployeeCvsPage.tsx (hunk 3 above)
Comment (INeizvestnaya) [reply src/pages/EmployeeCvsPage/EmployeeCvsPage.tsx:1]: It's ok, these pages are just not ready yet

File: src/components/UserProfileHeader/UserProfileHeader.tsx (hunk 7 above)
Comment (INeizvestnaya) [reply src/components/UserProfileHeader/UserProfileHeader.tsx:44]: This rule is useful that's why I don't disable it in config file. But I can't avoid the error here

File: src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx (hunk 5 above)
Comment (INeizvestnaya) [reply src/pages/EmployeeProfilePage/EmployeeProfilePage.tsx:5]: will be done soon

**Pull request merged**